### DISCUSSION TITLE
How to install this application in web server

### DISCUSSION THREAD

**Discussion by spy-way** [dyc3/opentogethertube#1812] [Q&A]
Please give me instructions about how to install this application in web server? ,and requirements?

**Comment by dyc3** [comment]
https://github.com/dyc3/opentogethertube/blob/master/docs/how-to-deploy.md

**Reply by spy-way** [reply]
I have cloud vps server,so can i install to vps server?

**Reply by dyc3** [reply]
Yeah, you can install whatever software you want on a vps server.

**Reply by spy-way** [reply]
How to install this application by using docker?

**Reply by dyc3** [reply]
That is explained in the linked document.

**Reply by spy-way** [reply]
![IMG_20241111_074406](https://github.com/user-attachments/assets/c7fd8e49-9231-42a6-abd8-4734b5887f2b)
I successfully completed your steps then run my website I got error like index page not load. Where is index page

**Reply by dyc3** [reply]
If you just did `docker compose up`, then OTT will be available on `http://localhost:8080` in your web browser. Note that you can't give that URL to other users because `localhost` refers to the machine you are on. The URL you give to users depends on how you are hosting OTT. There isn't really an "index page" in the traditional sense. Technically, the `index.html` is at `client/dist/index.html`, but it won't render properly if the server isn't serving the javascript.

> I got error like index page not load.

I can't help you troubleshoot errors like this without details. Can you show me any screenshots for this error? Browser console logs would also be helpful.

**Reply by spy-way** [reply]
I'm using vps server so I how I use localhost

**Reply by dyc3** [reply]
Replace `localhost` with whatever public IP your VPS is using.

**Reply by spy-way** [reply]
You mean .env file?

**Reply by dyc3** [reply]
If you are unable to determine the public IP of your VPS, you will have to ask your provider's tech support.

**Reply by spy-way** [reply]
I deploy the application in my server but page not loading I get  403 error,. Can you please help me I send you my server details

**Reply by dyc3** [reply]
Sorry, I can't manage your server for you. I can help you troubleshoot, but I need more information than just "it shows error." 403 doesn't mean anything to me. What url are you navigating to? What is the exact error you are getting? Does anything show up in the logs? Any screenshots?

**Reply by spy-way** [reply]
![Screenshot_20241112_050314](https://github.com/user-attachments/assets/40ed1c31-c610-4be8-941d-764aa1798f49)
![Screenshot_20241112_045812](https://github.com/user-attachments/assets/2303ea45-e5da-4e3f-85e3-e27c2840294e)
Oh ok , I got error log like nginx lndex problem

**Reply by dyc3** [reply]
Your nginx is not configured correctly. Have you adjusted the example nginx config to fit your deployment?

```nginx
map $http_upgrade $connection_upgrade {
    default upgrade;
    '' close;
}

server {
    listen 443 ssl http2;
    server_name example.com;

    # SSL configuration -- likely generated by certbot
    ssl_certificate /path/to/cert.pem;
    ssl_certificate_key /path/to/key.pem;

    location / {
        proxy_pass http://localhost:3000;
        proxy_http_version 1.1;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection "upgrade";
        proxy_set_header Host $host;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;
        proxy_set_header X-Forwarded-Host $host;
        proxy_cache_bypass $http_upgrade;
        proxy_pass_header Set-Cookie;

        # optional, but recommended if you have problem with clients disconnecting and reconnecting frequently
        proxy_connect_timeout 7d;
        proxy_read_timeout 7d;
        proxy_send_timeout 7d;

        proxy_socket_keepalive on;
    }
}
```

Make sure the `proxy_pass` url has the port you have configured for OTT, otherwise it will not work.

**Reply by spy-way** [reply]
Ok bro I try to change it , but I remove this server config then put your nginx config right? 
![Uploading 17313698841606550606344408982246.jpg…]()

**Reply by spy-way** [reply]
![Uploading 17313702033416746710612250029445.jpg…]()

**Reply by dyc3** [reply]
You also need to restart nginx for changes to be applied.

**Reply by spy-way** [reply]
I changed nginx config , then restart I got "Internal server error"

**Reply by dyc3** [reply]
Can you post your nginx config? Again, I can't really help if you don't give me information to work with.

**Reply by spy-way** [reply]
[Uploading my nginx.txt…]()

**Reply by spy-way** [reply]
![Uploading IMG_20241112_092313.jpg…]()

**Reply by spy-way** [reply]
![Uploading IMG_20241112_092356.jpg…]()

**Reply by spy-way** [reply]
https://ibb.co/jMvTrmV
https://ibb.co/g7nvscr
https://ibb.co/wWwLRBc
https://ibb.co/grDX9kV
https://ibb.co/kxm5Qzy
https://ibb.co/2PJNFNj
https://ibb.co/q1GrKH2

**Reply by spy-way** [reply]
Bro?

**Reply by dyc3** [reply]
> [Uploading my nginx.txt…](https://github.com/dyc3/opentogethertube/discussions/1812#discussioncomment-11220922)

FYI These links don't go to anything. You need to wait for the upload to finish before clicking reply.

From the screenshots you showed me, you aren't running the correct command to start the server. The `format` and `lint` scripts are dev tools. If you are using docker (which you should), the command you need to run is `docker compose up -d`.

**Reply by spy-way** [reply]
Oh ok . I send you my old nginx config you can put the correct line

**Reply by spy-way** [reply]
I try docker comment but that also not working

**Reply by dyc3** [reply]
If you've installed docker, it should just work. Please provide logs if it does not.

**Reply by spy-way** [reply]
Can you tell me the steps, I follow to deploy this app?

**Reply by dyc3** [reply]
[The steps are in this document](https://github.com/dyc3/opentogethertube/blob/master/docs/how-to-deploy.md), previously linked above. I am unable to help you and improve the instructions if you are unable to articulate what is confusing about those instructions.

**Reply by spy-way** [reply]
I deleted all my old deploy I started Newly

**Reply by spy-way** [reply]
![Screenshot_2024_1113_064834](https://github.com/user-attachments/assets/7de23389-cea0-48de-b462-793353187b1e)
I follow this steps only for deploy it's enough to run the project? And then it's enough step to run, we should use this "yarn install "command or shouldn't?

**Reply by spy-way** [reply]
![IMG_20241113_072312](https://github.com/user-attachments/assets/8bbb0e60-6bfd-499b-ba4e-0320fd00c8bc)

**Reply by spy-way** [reply]
![17314628919984433109735341566912](https://github.com/user-attachments/assets/4a38e049-23f0-498a-a945-00bae7bc24b8)

**Reply by spy-way** [reply]
![17314629502772057964146664121204](https://github.com/user-attachments/assets/65e2f1d7-5346-494b-b21b-0aeb706a1737)
![17314629654168748062820256763296](https://github.com/user-attachments/assets/6008fa74-c74c-4719-b8a6-1f1967ed884e)

**Reply by dyc3** [reply]
Have you filled out the `production.toml` config? The `req.logout` error only occurs if you left `session_secret` empty. You should use a password generator to generate an alphanumeric secret that is at least 80 characters long. You can also set the `SESSION_SECRET` environment variable for this.

**Reply by spy-way** [reply]
Bro morning I solved this issues, I put api key and session_secret key ..
